test/unit/routes/first-time/test-date-of-birth.js

```diff
@@ -1,73 +1,66 @@
+const sinon = require('sinon')
+require('sinon-bluebird')
+const ValidationError = require('../../../../app/services/errors/validation-error')
 
-var validationErrors
-
-var route = proxyquire(
-  '../../../../app/routes/first-time/new-eligibility/date-of-birth', {
-    '../../../services/validators/eligibility/date-of-birth-validator': function () { return validationErrors }
-  })
+var stubDateOfBirth
+var app
 
 describe('routes/first-time/new-eligibility/date-of-birth', function () {
   const ROUTE = '/first-time/new-eligibility'
 
-  var request
-  var dobDay = '01'
-  var dobMonth = '05'
-  var dobYear = '1955'
-  var dob = dateFormatter.buildFormatted(dobDay, dobMonth, dobYear)
-
   beforeEach(function () {
-    var app = express()
-    app.use(bodyParser.json())
-    mockViewEngine(app, '../../../app/views')
-    route(app)
-    request = supertest(app)
-    validationErrors = false
+    stubDateOfBirth = sinon.stub()
+    var route = proxyquire(
+      '../../../../app/routes/first-time/new-eligibility/date-of-birth', {
+        '../../../services/domain/date-of-birth': stubDateOfBirth
+      })
+
+    app = routeHelper.buildApp(route)
   })
 
   describe(`GET ${ROUTE}`, function () {
-    it('should respond with a 200', function (done) {
-      request
+    it('should respond with a 200', function () {
+      return supertest(app)
         .get(ROUTE)
         .expect(200)
-        .end(done)
     })
   })
 
   describe(`POST ${ROUTE}`, function () {
     describe('for over-sixteen date', function () {
-      it('should respond with a 302 and redirect /first-time/:dob', function (done) {
-        request
+      it('should respond with a 302 and redirect /first-time/new-eligibility/:dob', function () {
+        var dob = '1980-10-10'
+        stubDateOfBirth.returns({dobFormatted: dob, sixteenOrUnder: false})
+        return supertest(app)
           .post(ROUTE)
-          .send({
-            'dob-day': dobDay,
-            'dob-month': dobMonth,
-            'dob-year': dobYear
-          })
           .expect(302)
-          .expect('location', `${ROUTE}/${dob}`)
-          .end(done)
+          .expect(function () {
+            sinon.assert.calledOnce(stubDateOfBirth)
+          })
+          .expect('location', `/first-time/new-eligibility/${dob}`)
+      })
+
+      it('should respond with a 400 for invalid data', function () {
+        stubDateOfBirth.throws(new ValidationError({ 'dob': {} }))
+        return supertest(app)
+          .post(ROUTE)
+          .expect(400)
+          .expect(function () {
+            sinon.assert.calledOnce(stubDateOfBirth)
+          })
       })
     })
 
     describe('sixteen or under date', function () {
-      it('should respond with a 302 and redirect to /eligibility-fail', function (done) {
-        request
+      it('should respond with a 302 and redirect to /eligibility-fail', function () {
+        stubDateOfBirth.returns({sixteenOrUnder: true})
```

Comment:
you can stub a stub's methods to check route calls isSixteenOrUnder and return result